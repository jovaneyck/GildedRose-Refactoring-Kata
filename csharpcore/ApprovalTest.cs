using Xunit;
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;
using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalTests.Reporters.TestFrameworks;

namespace csharpcore
{
    [UseReporter(typeof(XUnit2Reporter))]
    public class ApprovalTest
    {
        [Fact]
        public void ThirtyDays()
        {
            var fakeoutput = new StringBuilder();
            Console.SetOut(new StringWriter(fakeoutput));
            Console.SetIn(new StringReader("a\n"));

            Program.Main(new string[] { });
            var output = fakeoutput.ToString();

            Approvals.Verify(output);
        }

        [Fact]
        public void AllCombinations()
        {
            var items = new List<Item>();

            

            ApprovalTests.Combinations.CombinationApprovals.VerifyAllCombinations(
                RunItem,
                new [] { "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros" },
                new[] { -1, 0, 1, 49, 50, 51 },
                new [] { -1, 0, 1, 5, 6, 7, 10, 11, 12 });
        }

        private string RunItem(string name, int quality, int sellIn)
        {
            var item = new Item{Name = name, Quality = quality, SellIn = sellIn};
            
            var app = new GildedRose(new List<Item> {item});
            app.UpdateQuality();

            return $"[{item.Name},{item.Quality},{item.SellIn}]";
        }
    }
}
