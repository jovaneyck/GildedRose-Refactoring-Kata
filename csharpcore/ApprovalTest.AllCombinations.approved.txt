[Aged Brie,-1,-1] => [Aged Brie,1,-2]
[Aged Brie,-1,0] => [Aged Brie,1,-1]
[Aged Brie,-1,1] => [Aged Brie,0,0]
[Aged Brie,-1,5] => [Aged Brie,0,4]
[Aged Brie,-1,6] => [Aged Brie,0,5]
[Aged Brie,-1,7] => [Aged Brie,0,6]
[Aged Brie,-1,10] => [Aged Brie,0,9]
[Aged Brie,-1,11] => [Aged Brie,0,10]
[Aged Brie,-1,12] => [Aged Brie,0,11]
[Aged Brie,0,-1] => [Aged Brie,2,-2]
[Aged Brie,0,0] => [Aged Brie,2,-1]
[Aged Brie,0,1] => [Aged Brie,1,0]
[Aged Brie,0,5] => [Aged Brie,1,4]
[Aged Brie,0,6] => [Aged Brie,1,5]
[Aged Brie,0,7] => [Aged Brie,1,6]
[Aged Brie,0,10] => [Aged Brie,1,9]
[Aged Brie,0,11] => [Aged Brie,1,10]
[Aged Brie,0,12] => [Aged Brie,1,11]
[Aged Brie,1,-1] => [Aged Brie,3,-2]
[Aged Brie,1,0] => [Aged Brie,3,-1]
[Aged Brie,1,1] => [Aged Brie,2,0]
[Aged Brie,1,5] => [Aged Brie,2,4]
[Aged Brie,1,6] => [Aged Brie,2,5]
[Aged Brie,1,7] => [Aged Brie,2,6]
[Aged Brie,1,10] => [Aged Brie,2,9]
[Aged Brie,1,11] => [Aged Brie,2,10]
[Aged Brie,1,12] => [Aged Brie,2,11]
[Aged Brie,49,-1] => [Aged Brie,50,-2]
[Aged Brie,49,0] => [Aged Brie,50,-1]
[Aged Brie,49,1] => [Aged Brie,50,0]
[Aged Brie,49,5] => [Aged Brie,50,4]
[Aged Brie,49,6] => [Aged Brie,50,5]
[Aged Brie,49,7] => [Aged Brie,50,6]
[Aged Brie,49,10] => [Aged Brie,50,9]
[Aged Brie,49,11] => [Aged Brie,50,10]
[Aged Brie,49,12] => [Aged Brie,50,11]
[Aged Brie,50,-1] => [Aged Brie,50,-2]
[Aged Brie,50,0] => [Aged Brie,50,-1]
[Aged Brie,50,1] => [Aged Brie,50,0]
[Aged Brie,50,5] => [Aged Brie,50,4]
[Aged Brie,50,6] => [Aged Brie,50,5]
[Aged Brie,50,7] => [Aged Brie,50,6]
[Aged Brie,50,10] => [Aged Brie,50,9]
[Aged Brie,50,11] => [Aged Brie,50,10]
[Aged Brie,50,12] => [Aged Brie,50,11]
[Aged Brie,51,-1] => [Aged Brie,51,-2]
[Aged Brie,51,0] => [Aged Brie,51,-1]
[Aged Brie,51,1] => [Aged Brie,51,0]
[Aged Brie,51,5] => [Aged Brie,51,4]
[Aged Brie,51,6] => [Aged Brie,51,5]
[Aged Brie,51,7] => [Aged Brie,51,6]
[Aged Brie,51,10] => [Aged Brie,51,9]
[Aged Brie,51,11] => [Aged Brie,51,10]
[Aged Brie,51,12] => [Aged Brie,51,11]
[Backstage passes to a TAFKAL80ETC concert,-1,-1] => [Backstage passes to a TAFKAL80ETC concert,0,-2]
[Backstage passes to a TAFKAL80ETC concert,-1,0] => [Backstage passes to a TAFKAL80ETC concert,0,-1]
[Backstage passes to a TAFKAL80ETC concert,-1,1] => [Backstage passes to a TAFKAL80ETC concert,2,0]
[Backstage passes to a TAFKAL80ETC concert,-1,5] => [Backstage passes to a TAFKAL80ETC concert,2,4]
[Backstage passes to a TAFKAL80ETC concert,-1,6] => [Backstage passes to a TAFKAL80ETC concert,1,5]
[Backstage passes to a TAFKAL80ETC concert,-1,7] => [Backstage passes to a TAFKAL80ETC concert,1,6]
[Backstage passes to a TAFKAL80ETC concert,-1,10] => [Backstage passes to a TAFKAL80ETC concert,1,9]
[Backstage passes to a TAFKAL80ETC concert,-1,11] => [Backstage passes to a TAFKAL80ETC concert,0,10]
[Backstage passes to a TAFKAL80ETC concert,-1,12] => [Backstage passes to a TAFKAL80ETC concert,0,11]
[Backstage passes to a TAFKAL80ETC concert,0,-1] => [Backstage passes to a TAFKAL80ETC concert,0,-2]
[Backstage passes to a TAFKAL80ETC concert,0,0] => [Backstage passes to a TAFKAL80ETC concert,0,-1]
[Backstage passes to a TAFKAL80ETC concert,0,1] => [Backstage passes to a TAFKAL80ETC concert,3,0]
[Backstage passes to a TAFKAL80ETC concert,0,5] => [Backstage passes to a TAFKAL80ETC concert,3,4]
[Backstage passes to a TAFKAL80ETC concert,0,6] => [Backstage passes to a TAFKAL80ETC concert,2,5]
[Backstage passes to a TAFKAL80ETC concert,0,7] => [Backstage passes to a TAFKAL80ETC concert,2,6]
[Backstage passes to a TAFKAL80ETC concert,0,10] => [Backstage passes to a TAFKAL80ETC concert,2,9]
[Backstage passes to a TAFKAL80ETC concert,0,11] => [Backstage passes to a TAFKAL80ETC concert,1,10]
[Backstage passes to a TAFKAL80ETC concert,0,12] => [Backstage passes to a TAFKAL80ETC concert,1,11]
[Backstage passes to a TAFKAL80ETC concert,1,-1] => [Backstage passes to a TAFKAL80ETC concert,0,-2]
[Backstage passes to a TAFKAL80ETC concert,1,0] => [Backstage passes to a TAFKAL80ETC concert,0,-1]
[Backstage passes to a TAFKAL80ETC concert,1,1] => [Backstage passes to a TAFKAL80ETC concert,4,0]
[Backstage passes to a TAFKAL80ETC concert,1,5] => [Backstage passes to a TAFKAL80ETC concert,4,4]
[Backstage passes to a TAFKAL80ETC concert,1,6] => [Backstage passes to a TAFKAL80ETC concert,3,5]
[Backstage passes to a TAFKAL80ETC concert,1,7] => [Backstage passes to a TAFKAL80ETC concert,3,6]
[Backstage passes to a TAFKAL80ETC concert,1,10] => [Backstage passes to a TAFKAL80ETC concert,3,9]
[Backstage passes to a TAFKAL80ETC concert,1,11] => [Backstage passes to a TAFKAL80ETC concert,2,10]
[Backstage passes to a TAFKAL80ETC concert,1,12] => [Backstage passes to a TAFKAL80ETC concert,2,11]
[Backstage passes to a TAFKAL80ETC concert,49,-1] => [Backstage passes to a TAFKAL80ETC concert,0,-2]
[Backstage passes to a TAFKAL80ETC concert,49,0] => [Backstage passes to a TAFKAL80ETC concert,0,-1]
[Backstage passes to a TAFKAL80ETC concert,49,1] => [Backstage passes to a TAFKAL80ETC concert,50,0]
[Backstage passes to a TAFKAL80ETC concert,49,5] => [Backstage passes to a TAFKAL80ETC concert,50,4]
[Backstage passes to a TAFKAL80ETC concert,49,6] => [Backstage passes to a TAFKAL80ETC concert,50,5]
[Backstage passes to a TAFKAL80ETC concert,49,7] => [Backstage passes to a TAFKAL80ETC concert,50,6]
[Backstage passes to a TAFKAL80ETC concert,49,10] => [Backstage passes to a TAFKAL80ETC concert,50,9]
[Backstage passes to a TAFKAL80ETC concert,49,11] => [Backstage passes to a TAFKAL80ETC concert,50,10]
[Backstage passes to a TAFKAL80ETC concert,49,12] => [Backstage passes to a TAFKAL80ETC concert,50,11]
[Backstage passes to a TAFKAL80ETC concert,50,-1] => [Backstage passes to a TAFKAL80ETC concert,0,-2]
[Backstage passes to a TAFKAL80ETC concert,50,0] => [Backstage passes to a TAFKAL80ETC concert,0,-1]
[Backstage passes to a TAFKAL80ETC concert,50,1] => [Backstage passes to a TAFKAL80ETC concert,50,0]
[Backstage passes to a TAFKAL80ETC concert,50,5] => [Backstage passes to a TAFKAL80ETC concert,50,4]
[Backstage passes to a TAFKAL80ETC concert,50,6] => [Backstage passes to a TAFKAL80ETC concert,50,5]
[Backstage passes to a TAFKAL80ETC concert,50,7] => [Backstage passes to a TAFKAL80ETC concert,50,6]
[Backstage passes to a TAFKAL80ETC concert,50,10] => [Backstage passes to a TAFKAL80ETC concert,50,9]
[Backstage passes to a TAFKAL80ETC concert,50,11] => [Backstage passes to a TAFKAL80ETC concert,50,10]
[Backstage passes to a TAFKAL80ETC concert,50,12] => [Backstage passes to a TAFKAL80ETC concert,50,11]
[Backstage passes to a TAFKAL80ETC concert,51,-1] => [Backstage passes to a TAFKAL80ETC concert,0,-2]
[Backstage passes to a TAFKAL80ETC concert,51,0] => [Backstage passes to a TAFKAL80ETC concert,0,-1]
[Backstage passes to a TAFKAL80ETC concert,51,1] => [Backstage passes to a TAFKAL80ETC concert,51,0]
[Backstage passes to a TAFKAL80ETC concert,51,5] => [Backstage passes to a TAFKAL80ETC concert,51,4]
[Backstage passes to a TAFKAL80ETC concert,51,6] => [Backstage passes to a TAFKAL80ETC concert,51,5]
[Backstage passes to a TAFKAL80ETC concert,51,7] => [Backstage passes to a TAFKAL80ETC concert,51,6]
[Backstage passes to a TAFKAL80ETC concert,51,10] => [Backstage passes to a TAFKAL80ETC concert,51,9]
[Backstage passes to a TAFKAL80ETC concert,51,11] => [Backstage passes to a TAFKAL80ETC concert,51,10]
[Backstage passes to a TAFKAL80ETC concert,51,12] => [Backstage passes to a TAFKAL80ETC concert,51,11]
[Sulfuras, Hand of Ragnaros,-1,-1] => [Sulfuras, Hand of Ragnaros,-1,-1]
[Sulfuras, Hand of Ragnaros,-1,0] => [Sulfuras, Hand of Ragnaros,-1,0]
[Sulfuras, Hand of Ragnaros,-1,1] => [Sulfuras, Hand of Ragnaros,-1,1]
[Sulfuras, Hand of Ragnaros,-1,5] => [Sulfuras, Hand of Ragnaros,-1,5]
[Sulfuras, Hand of Ragnaros,-1,6] => [Sulfuras, Hand of Ragnaros,-1,6]
[Sulfuras, Hand of Ragnaros,-1,7] => [Sulfuras, Hand of Ragnaros,-1,7]
[Sulfuras, Hand of Ragnaros,-1,10] => [Sulfuras, Hand of Ragnaros,-1,10]
[Sulfuras, Hand of Ragnaros,-1,11] => [Sulfuras, Hand of Ragnaros,-1,11]
[Sulfuras, Hand of Ragnaros,-1,12] => [Sulfuras, Hand of Ragnaros,-1,12]
[Sulfuras, Hand of Ragnaros,0,-1] => [Sulfuras, Hand of Ragnaros,0,-1]
[Sulfuras, Hand of Ragnaros,0,0] => [Sulfuras, Hand of Ragnaros,0,0]
[Sulfuras, Hand of Ragnaros,0,1] => [Sulfuras, Hand of Ragnaros,0,1]
[Sulfuras, Hand of Ragnaros,0,5] => [Sulfuras, Hand of Ragnaros,0,5]
[Sulfuras, Hand of Ragnaros,0,6] => [Sulfuras, Hand of Ragnaros,0,6]
[Sulfuras, Hand of Ragnaros,0,7] => [Sulfuras, Hand of Ragnaros,0,7]
[Sulfuras, Hand of Ragnaros,0,10] => [Sulfuras, Hand of Ragnaros,0,10]
[Sulfuras, Hand of Ragnaros,0,11] => [Sulfuras, Hand of Ragnaros,0,11]
[Sulfuras, Hand of Ragnaros,0,12] => [Sulfuras, Hand of Ragnaros,0,12]
[Sulfuras, Hand of Ragnaros,1,-1] => [Sulfuras, Hand of Ragnaros,1,-1]
[Sulfuras, Hand of Ragnaros,1,0] => [Sulfuras, Hand of Ragnaros,1,0]
[Sulfuras, Hand of Ragnaros,1,1] => [Sulfuras, Hand of Ragnaros,1,1]
[Sulfuras, Hand of Ragnaros,1,5] => [Sulfuras, Hand of Ragnaros,1,5]
[Sulfuras, Hand of Ragnaros,1,6] => [Sulfuras, Hand of Ragnaros,1,6]
[Sulfuras, Hand of Ragnaros,1,7] => [Sulfuras, Hand of Ragnaros,1,7]
[Sulfuras, Hand of Ragnaros,1,10] => [Sulfuras, Hand of Ragnaros,1,10]
[Sulfuras, Hand of Ragnaros,1,11] => [Sulfuras, Hand of Ragnaros,1,11]
[Sulfuras, Hand of Ragnaros,1,12] => [Sulfuras, Hand of Ragnaros,1,12]
[Sulfuras, Hand of Ragnaros,49,-1] => [Sulfuras, Hand of Ragnaros,49,-1]
[Sulfuras, Hand of Ragnaros,49,0] => [Sulfuras, Hand of Ragnaros,49,0]
[Sulfuras, Hand of Ragnaros,49,1] => [Sulfuras, Hand of Ragnaros,49,1]
[Sulfuras, Hand of Ragnaros,49,5] => [Sulfuras, Hand of Ragnaros,49,5]
[Sulfuras, Hand of Ragnaros,49,6] => [Sulfuras, Hand of Ragnaros,49,6]
[Sulfuras, Hand of Ragnaros,49,7] => [Sulfuras, Hand of Ragnaros,49,7]
[Sulfuras, Hand of Ragnaros,49,10] => [Sulfuras, Hand of Ragnaros,49,10]
[Sulfuras, Hand of Ragnaros,49,11] => [Sulfuras, Hand of Ragnaros,49,11]
[Sulfuras, Hand of Ragnaros,49,12] => [Sulfuras, Hand of Ragnaros,49,12]
[Sulfuras, Hand of Ragnaros,50,-1] => [Sulfuras, Hand of Ragnaros,50,-1]
[Sulfuras, Hand of Ragnaros,50,0] => [Sulfuras, Hand of Ragnaros,50,0]
[Sulfuras, Hand of Ragnaros,50,1] => [Sulfuras, Hand of Ragnaros,50,1]
[Sulfuras, Hand of Ragnaros,50,5] => [Sulfuras, Hand of Ragnaros,50,5]
[Sulfuras, Hand of Ragnaros,50,6] => [Sulfuras, Hand of Ragnaros,50,6]
[Sulfuras, Hand of Ragnaros,50,7] => [Sulfuras, Hand of Ragnaros,50,7]
[Sulfuras, Hand of Ragnaros,50,10] => [Sulfuras, Hand of Ragnaros,50,10]
[Sulfuras, Hand of Ragnaros,50,11] => [Sulfuras, Hand of Ragnaros,50,11]
[Sulfuras, Hand of Ragnaros,50,12] => [Sulfuras, Hand of Ragnaros,50,12]
[Sulfuras, Hand of Ragnaros,51,-1] => [Sulfuras, Hand of Ragnaros,51,-1]
[Sulfuras, Hand of Ragnaros,51,0] => [Sulfuras, Hand of Ragnaros,51,0]
[Sulfuras, Hand of Ragnaros,51,1] => [Sulfuras, Hand of Ragnaros,51,1]
[Sulfuras, Hand of Ragnaros,51,5] => [Sulfuras, Hand of Ragnaros,51,5]
[Sulfuras, Hand of Ragnaros,51,6] => [Sulfuras, Hand of Ragnaros,51,6]
[Sulfuras, Hand of Ragnaros,51,7] => [Sulfuras, Hand of Ragnaros,51,7]
[Sulfuras, Hand of Ragnaros,51,10] => [Sulfuras, Hand of Ragnaros,51,10]
[Sulfuras, Hand of Ragnaros,51,11] => [Sulfuras, Hand of Ragnaros,51,11]
[Sulfuras, Hand of Ragnaros,51,12] => [Sulfuras, Hand of Ragnaros,51,12]
